auth/README.md

```diff
@@ -1,333 +1,269 @@
 # Authentication
 
-In order to get data from the Vimeo API, we need to generate an access token. There are two types of access tokens - basic and authenticated. It can be obtained by using the [Authenticator](./src/main/java/com/vimeo/networking2/Authenticator.kt) interface. It provides methods to authenticate via email, Google or Facebook. These methods will provide you with an access token via a callback.
+In order to get data from the Vimeo API, we need to generate an access token. There are two types of access tokens - basic and authenticated. It can be obtained by using the [Authenticator](./src/main/java/com/vimeo/networking2/Authenticator.kt) interface. It provides methods to authenticate via code grant, email login, Google login, or Facebook login. These methods will provide you with an access token via a callback.
 
 ## Basic Authentication
 
-A basic access token can be obtained by using your client id, client secret and access scopes. A client id and client secret can created by going to [Create an App](https://developer.vimeo.com/).
+A basic access token can be obtained by using your client ID, client secret and access scopes. A client id and client secret can created by going to [Create an App](https://developer.vimeo.com/).
 
-After you get your client credentials, create an instance of [ServerConfig](../api/src/main/java/com/vimeo/networking2/config/ServerConfig.kt) as  follows:
+After you get your client credentials, create an instance of [VimeoApiConfiguration](../api-core/src/main/java/com/vimeo/networking2/config/VimeoApiConfiguration.kt) as  follows:
 
 ```kotlin
-
 // Kotlin
-val serverConfig = ServerConfig(CLIENT_ID, CLIENT_SECRET)
-
+val configuration = VimeoApiConfiguration.Builder(CLIENT_ID, CLIENT_SECRET, listOf(ScopeType.PUBLIC, ScopeType.PRIVATE))
+    .build()
 ```
 
 ```java
-
 // Java
-ServerConfig serverConfig = new ServerConfig(CLIENT_ID, CLIENT_SECRET);
-
+final List<ScopeType> scopes = Arrays.asList(ScopeType.PUBLIC, ScopeType.PRIVATE);
+final VimeoApiConfiguration configuration = new VimeoApiConfiguration.Builder(CLIENT_ID, CLIENT_SECRET, scopes).build();
 ```
 
-In order to authenticate with your client credentials, create an instance of [Authenticator](./src/main/java/com/vimeo/networking2/Authenticator.kt) and pass your `ServerConfig` object to it.
+In order to authenticate with your client credentials, create an instance of [Authenticator](../auth/src/main/java/com/vimeo/networking2/Authenticator.kt) and pass your `VimeoApiConfiguration` object to it.
 
 ```kotlin
+// Kotlin (new instance)
+val authenticator = Authenticator(vimeoApiConfiguration)
 
-// Kotlin
-val authenticator = Authenticator.create(serverConfig)
-
+// Kotlin (singleton)
+Authenticator.initialize(vimeoApiConfiguration)
 ```
 
 ```java
+// Java (new instance)
```

Comment:
I think I'll just remove them then as I agree that consistency is important. Adding java docs to the main readme will just get way too verbose. It would be nice if markdown supported like an alternate language block so people could toggle, but having 2 blocks for everything is just a bit much I think.